File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        unordered_map<int, int> counter;
        for (int num : arr) {
            counter[num]++;
        }
        for (const auto& entry : counter) {
            if (entry.second > arr.size() / 4) {
                return entry.first;
            }
        }
        return 0;
    }
};

File: LeetCode/1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        Map<Integer, Integer> counter = new HashMap<Integer, Integer>();
        for (int num : arr) {
            counter.put(num, counter.getOrDefault(num, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> entry : counter.entrySet()) {
            if (entry.getValue() > arr.length / 4) {
                return entry.getKey();
            }
        }
        return 0;
    }
}